examples/embeddable_examples/public/app/state_management_example/state_management_example.tsx

```diff
@@ -0,0 +1,175 @@
+      subscription.unsubscribe();
+    };
+  }, [bookApi]);
+
+  return (
+    <div>
+      <EuiCallOut>
+        <p>
+          Each embeddable manages its own state. The page is only responsible for persisting and
+          providing the last persisted state to the embeddable.
+        </p>
+
+        <p>
+          The page renders the embeddable with <strong>ReactEmbeddableRenderer</strong> component.
+          On mount, ReactEmbeddableRenderer component calls{' '}
+          <strong>pageApi.getSerializedStateForChild</strong> to get the last saved state.
+          ReactEmbeddableRenderer component then calls{' '}
+          <strong>pageApi.getRuntimeStateForChild</strong> to get the last session&apos;s unsaved
+          changes. ReactEmbeddableRenderer merges last saved state with unsaved changes and passes
+          the merged state to the embeddable factory. ReactEmbeddableRender passes the embeddableApi
+          to the page by calling <strong>onApiAvailable</strong>.
+        </p>
+
+        <p>
+          The page subscribes to <strong>embeddableApi.unsavedChanges</strong> to receive embeddable
+          unsaved changes. The page persists unsaved changes in session storage. The page provides
+          unsaved changes to the embeddable with <strong>pageApi.getRuntimeStateForChild</strong>.
+        </p>
+
+        <p>
+          The page gets embeddable state by calling <strong>embeddableApi.serializeState</strong>.
+          The page persists embeddable state in session storage. The page provides last saved state
+          to the embeddable with <strong>pageApi.getSerializedStateForChild</strong>.
+        </p>
+
+        <p>
+          <EuiButtonEmpty
+            color={'warning'}
+            onClick={() => {
+              lastSavedStateSessionStorage.clear();
+              unsavedChangesSessionStorage.clear();
+              window.location.reload();
+            }}
+          >
+            Reset example
+          </EuiButtonEmpty>
+        </p>
+      </EuiCallOut>
+
+      <EuiSpacer size="m" />
+
+      <EuiFlexGroup justifyContent="flexEnd">
+        {hasUnsavedChanges && (
+          <>
+            <EuiFlexItem grow={false}>
+              <EuiBadge color="warning">Unsaved changes</EuiBadge>
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <EuiButtonEmpty
+                disabled={isSaving || !bookApi}
+                onClick={() => {
+                  bookApi?.resetUnsavedChanges?.();
+                }}
+              >
+                Reset
+              </EuiButtonEmpty>
+            </EuiFlexItem>
+          </>
+        )}
+        <EuiFlexItem grow={false}>
+          <EuiButton
+            disabled={isSaving || !hasUnsavedChanges}
+            onClick={async () => {
+              if (!bookApi) {
+                return;
+              }
+
+              setIsSaving(true);
+              const bookSerializedState = await bookApi.serializeState();
```

Comment:
@nreese I was wondering if you should do `await bookApi.serializeState().rawState;` here. Do we really want `rawState` to be saved in the sessionStorage?

<img width="1389" alt="Screenshot 2024-09-11 at 22 33 06" src="https://github.com/user-attachments/assets/031ab847-d475-4210-a5c2-e27c77c5dcc1">

Reply:
Both rawState and references are needed. This example does not use references but you can not throw away information. For example, if the embeddable used a map embeddable instead, then references would provide data view saved object ids. The state structure should be opaque to the page. The page is only required to provide what it was given back to the embeddable on page load.

Reply:
Then my confusion comes from the fact that I didn't see any `rawState` in the dashboard SO. In your example you use session Storage to store your data, in the dashboard page data gets stored in the dashboard SO. That's why I got confused why there is `rawState` (and references) in sessionStorage and not in SO. 

<img width="1398" alt="Screenshot 2024-09-11 at 22 08 03" src="https://github.com/user-attachments/assets/b0a862f2-514d-434a-9aa3-5605ec8b160f">

Reply:
The best way to see where dashboard saves embeddable state is to create a dashboard with a single links panel. After saving the dashboard, inspect the saved object in the saved object management page. Dashboard extracts references from each embeddable and stores them  in `references`. Dashboard stores embeddable rawState under `panelsJSON`. If you parse `panelsJSON`, its an array of objects. Raw state is stored under `embeddableConfig` key for each object. This is hard to follow, which is why we are creating the examples. That gives us a clean space to make clear examples without all the noise and legacy of Dashboard.

<img width="800" alt="Screenshot 2024-09-12 at 7 20 10 AM" src="https://github.com/user-attachments/assets/c447cd11-74be-41c5-9ad9-28aec140c692">

Reply:
You may ask, why does dashboard do all of this reference extraction. It was done so that saved objects can track and expose dependencies on other saved objects.

Reply:
Thanks for the explanation! My main concern was why the `rawState` key is stored in the session storage in your example where as the dashboard SO panelsJSON in my screenshot above doesn't contain the key `rawState`. Is it correct that your example saves the `rawState` key in the `kibana.examples.embeddables.stateManagementExample.unsavedChanges` and `kibana.examples.embeddables.stateManagementExample.savedState` sessionStorage?

Reply:
So to put it differently, if I want to store the rawState in our `investigation` SO, do I need to store the `rawState key`? Or am I flexible to call it as I want (`items` in our case).  In the dashboard app rawState is stored as `panelsJSON` and that's where I got confused, if there's a convention on the key I should use to save the rawState.

Reply:
`kibana.examples.embeddables.stateManagementExample.savedState` stores serialized state, which is of the type `SerializedPanelState<BookSerializedState>` and looks like `{ references: [], rawState: {} as BookSerializedState }`

`kibana.examples.embeddables.stateManagementExample.unsavedChanges` stores runtime state, which is of the type `Partial<BookRuntimeState>`

Reply:
> So to put it differently, if I want to store the rawState in our investigation SO, do I need to store the rawState key? Or am I flexible to call it as I want (items in our case). In the dashboard app rawState is stored as panelsJSON and that's where I got confused, if there's a convention on the key I should use to save the rawState.

You can store it however you like, you just need to return in the shape `{ references: [], rawState: {} }`. You can not loose the references piece.